x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsRelocationIntegTests.java

```diff
@@ -123,9 +123,7 @@ public void testRelocationWaitsForPreWarm() throws Exception {
             .shardRecoveryStates()
             .get(restoredIndex)
             .stream()
-            // filter for relocations that are not in stage FINALIZE (they could end up in this stage without progress for good if the
-            // target node does not have enough cache space available to hold the primary completely
-            .filter(recoveryState -> recoveryState.getSourceNode() != null && recoveryState.getStage() != RecoveryState.Stage.FINALIZE)
+            .filter(recoveryState -> recoveryState.getSourceNode() != null)
```

Comment:
Maybe we could check at the end that the pre-warming phase ended?

Reply:
Yea we'd have to do something in that direction, otherwise https://github.com/elastic/elasticsearch/pull/67697/files#diff-be91049c27fa011e281944d790285ad97bbd72466aeeed55525c5b99061c9e13L113 will trip for small cache sizes that are randomly chosen in the parent test suite because we will get stuck in `FINALIZE` forever right?